include/glow/Optimizer/FunctionPass.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_FUNCTIONPASS_H
+#define GLOW_OPTIMIZER_FUNCTIONPASS_H
+
+namespace glow {
+
+class FunctionPass {
```

Comment:
Some designs of pass managers do not take `Function` as an explicit constructor argument of their passes. Instead they are injected by the pass manager via some APIs exposed by the pass. See e.g. https://github.com/apple/swift/blob/master/include/swift/SILOptimizer/PassManager/Transforms.h

I don't have a strong opinion about this.

Reply:
Another interesting bit in the example above is the way how passes are mass-defined by means of a def file and macro meta-programming:
https://github.com/apple/swift/blob/master/include/swift/SILOptimizer/PassManager/Passes.def

Reply:
I'm not sure I see the benefit of injection for now. I do like the macro meta-programming, I think it will be useful into the future when integrating into the pass manager and allowing backends to opt in/out of passes, or specify their own pass pipeline.